system/Model.php

```diff
@@ -275,7 +275,18 @@ protected function doInsert(array $data)
             $builder->set($key, $val, $escape[$key] ?? null);
         }
 
-        $result = $builder->insert();
+        if (! $this->prohibitInsertEmpty && empty($data)) {
+            $table = $this->db->protectIdentifiers($this->table, true, null, false);
+            if ($this->db->getPlatform() === 'MySQLi') {
+                $sql = 'INSERT INTO ' . $table . ' VALUES ()';
+            } else {
+                $sql = 'INSERT INTO ' . $table . ' DEFAULT VALUES';
```

Comment:
I'm not familiar with this but tests seem to vet it.

Reply:
Oracle does not work with it. 😓